src/Symfony/Component/Translation/Translator.php

```diff
@@ -91,6 +91,10 @@ class Translator implements TranslatorInterface, TranslatorBagInterface, LocaleA
      */
     public function __construct(string $locale, MessageFormatterInterface $formatter = null, string $cacheDir = null, bool $debug = false, array $cacheVary = [])
     {
+        if ('' === $locale) {
+            trigger_deprecation('symfony/translator', '5.2', sprintf('Passing "" as the $locale argument to %s::%s() is deprecated, it will throw an InvalidArgumentException in version 6.0.', static::class, __METHOD__));
```

Comment:
I'm not sure this case makes sense. It conflicts, from a logical pov, with the change in getLocale(). It's one or the other to me, isn't it?

Reply:
I think it depends. 

If we say injecting an empty locale in the constructor is ok, because we make sure that the default locale will be returned in `getLocale()` anyway, then yes a deprecation is not needed. Only Tests need some adjustments then.

But if we say an empty locale is not valid, we could validate it in `assertValidLocale()` and also simplify getLocale() by removing the `\?: Locale::getDefault()`. But this would be a BC break if we do it right now, so I only  introduced the deprecation for now and would do the adjustments of `assertValidLocale()` and `getLocale()` in version 6.

I personally would prefer the second approach, but I also might miss something :-)

Reply:
we should patch `assertValidLocale` IMHO. deprecating an empty string there, throwing in 6.0

given the locale is not a nullable in this case, there's no need for fallback to `\Locale::getDefault()` as well

thus the deprecated case (an empty string) keeps working in 5.x, as currently that's a BC break.

we should proboably call `$this->setLocale($locale)` at construct (to ensure validation), meaning the deprecation will automatically happen as well.

Reply:
I would personnaly take the other way, for consistency with TranslatorTrait, and because that's what provides the most power to users (they can decide to rely on `\Locale::getDefault()`)

Reply:
it can also take a nullable again :+1: 

but yes, `?: getDefault()` works, at most it hides `"0"` sneaking into the system. But it's consistent behavior with the trait, i agree.

IMHO the service differs from the trait, in a way doing `new Translator(Locale::getDefault()` makes sense, otherwise we have to document the empty string behavior as well i figured.

The trait needs a hardcoded fallback mechanism, due the lack of a constructor.

Reply:
we still need a deprecation here (and in setLocale) then

```php
if (!$locale) {
    trigger_deprecation('will be Locale::getDefault() (currently "...") in 6.0');
    // $locale = Locale::getDefault()
}
```

Reply:
I don't see the point of the deprecation. Let's move on.